tests/lax_control_flow_test.py

```diff
@@ -3311,6 +3311,14 @@ def g(x):
     eqn_jaxpr = jaxpr.eqns[0].params["jaxpr"]
     self.assertIn("debug_callback", [e.primitive.name for e in eqn_jaxpr.eqns])
 
+  def test_scan_input_to_output_forwarding(self):
```

Comment:
We should write really thorough tests, maybe some combinatorially generated ones, like these:

https://github.com/jax-ml/jax/blob/f5a9d460723b417ca2032057f8b2ef5ad9e6fdf9/tests/lax_control_flow_test.py#L3196